.ci/tox_envs/update.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+THIS_DIRECTORY="$(dirname "$0")"
+ENVS=(flake8 flake8_imports)
+
+for env in "${ENVS[@]}"
+do
+        cd "$THIS_DIRECTORY/$env"
```

Comment:
For what I can see, this isn't actually updating the requirements, just creating `Pipfile` and `requirements.txt`.
Should there be a:
```
pipenv update
```
here?

Reply:
I was imaging more of less just keeping the Pipfile un-pinned except when it is absolutely needed (e.g. Twill, pysam, etc...) - the pinning would happen in the requirements and lock file creation here. Re-generating the lock file rebuilds a ``Pipfile.lock`` (and associated requirements files) with the newest compatible versions of the libraries in the ``Pipfile``.  If we generated new lock and requirements files and we discover they are incompatible during testing - we add version restrictions in the Pipfile and then re-run and generate the appropriate lock and requirement files.

I can't find good documentation on ``pipenv update`` and it doesn't work for me locally. I think it is for installing newer updates to the dependencies not for updating the Pipfile right? Even if it was for updating the ``Pipfile``, I think for the above workflow it would be bad to update the locked dependencies automatically since we are only locking things in the ``Pipfile`` if we know we need to lock them.

I did notice that the ``Pipfile.lock`` files weren't added since that pattern is our ``.gitignore``.  I've added them to https://github.com/galaxyproject/galaxy/pull/4903 but I can back port them here too - but since we are still just using the generate requirements files with pip - it doesn't seem important to track the lock files at this time.

Reply:
Thanks for the explanation, I assumed that `pipenv lock` would create the lock file according to the versions installed in the virtualenv, not also update packages if they're not pinned.
It seems the documentation needs to be clearer on both commands.